content/developer/tutorials/restrict_data_access.rst

```diff
@@ -355,21 +355,15 @@ Explicit security checks can be performed by:
   specific models or records.
 * Checking that the current user has specific groups hard-coded to allow or deny
   an operation (``self.env.user.has_group``).
-* Calling the ``check_access_rights(operation)`` method on a recordset, this
-  verifies whether the current user has access to the model itself.
-* Calling ``check_access_rule(operations)`` on a non-empty recordset, this
-  verifies that the current user is allowed to perform the operation on *every*
-  record of the set.
-
-.. warning:: Checking access rights and checking record rules are separate
-             operations, if you're checking record rules you usually want to
-             also check access rights beforehand.
+* Calling ``check_access(operations)`` on a recordset, this verifies that the
+  current user is allowed to perform the operation on *every* record of the set.
+  As a special case, when the recordset is empty, it verifies that the current
+  user has some permission to perform the operation on the model in general.
```

Comment:
I'm not sure to understand what you mean by "some permission"? As in "access through some (at least one) ACL"?

Reply:
That's it.  Some permission means some ACL.